File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        queue<char> q;
        unordered_map<char, int> m;

        for(char c: s) {
            q.push(c);
            m[c]++;
        }

        while(!q.empty()) {
            char c = q.front();
            q.pop();

            if(m[c] == 1) return s.size() - q.size() - 1;
        }

        return -1;
    }
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        m = defaultdict(int)
        for c in s:
            m[c] += 1
        
        for i, c in enumerate(list(s)):
            if m[c] == 1:
                return i
        return -1